### PR TITLE
Added Zachary's Quantitative Analysis portion and Aanchal's Efficient Frontier portion.

### PR DESCRIPTION
**Pull request by aakhanna-1909**
Added Zachary's Quantitative Analysis portion and Aanchal's Efficient Frontier portion.

### CODE REVIEW

**Review by BubbaHexinator — APPROVED** [review]

File: modules/sharpe_ratio.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+def sharpe_ratio(portfolio, tickers):
+  sharpe_ratio_list = []
+  ticker_list = []
+  
+  for ticker in tickers:
+    ticker_list.append(ticker)
+    ticker_daily_returns = portfolio.loc[:, (ticker, "daily_returns")]
+    sharpe_ratio_list.append(qs.stats.sharpe(ticker_daily_returns))
+  
+  sharpe_ratio_df = pd.DataFrame([ticker_list, sharpe_ratio_list]).T
```
Comment (mmarovich) [inline modules/sharpe_ratio.py:16]:
```suggestion
  # Needed to call .T method to transpose data properly
  sharpe_ratio_df = pd.DataFrame([ticker_list, sharpe_ratio_list]).T
```

File: modules/volatility_analysis.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import numpy as np
+import pandas as pd
+import seaborn as sns
+number_trading_days = 252
+
+def volatility_analysis(portfolio, tickers):
+  each_ticker_volatility = []
+  each_ticker = []
+  for ticker in tickers:
```
Comment (mmarovich) [inline modules/volatility_analysis.py]:
```suggestion
  each_ticker = []
  
  for ticker in tickers:
```

File: modules/volatility_analysis.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    each_ticker.append(ticker)
+    daily_returns = portfolio.loc[:, (ticker, "daily_returns")]
+    each_ticker_volatility.append(daily_returns.std() * np.sqrt(number_trading_days))
+  
+  volatility_df = pd.DataFrame([each_ticker, each_ticker_volatility]).T
+  cols = ["Stocks", "Volatility"]
+  volatility_df.columns=cols
+  volatility_df.sort_values(by="Volatility", inplace=True)
+  sns.barplot(data=volatility_df, x="Stocks", y="Volatility", palette="flare")
+  return(volatility_df)
```
Comment (mmarovich) [inline modules/volatility_analysis.py]:
```suggestion
  display(volatility_df)
```

File: modules/sp500.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    # Setting index to date for better index/date formatting
+    sp500_df.index = sp500_df.index.date
+
+    # Calculate "daily_returns" for S&P 500
+    sp500_df["daily_returns"] = sp500_df["close"].pct_change()
+
+    
+    # Return the S&P 500 dataframe
+
+    return sp500_df
```
Comment (mmarovich) [inline modules/sp500.py:43]: I think we can use code that has already been written for the "SPY" ticker.  Let's use alpaca call function to get the dataframe, and we can use the calculate returns function to get the returns, and we can retrieve this information at the start of our app, to have it ready when we need it.

File: modules/beta.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import seaborn as sns
+import pandas as pd
+
+def beta(portfolio, tickers, sp500_df):
+    beta_list = []
+    ticker_list = []
+    for ticker in tickers:
```
Comment (mmarovich) [inline modules/beta.py]:
```suggestion
    ticker_list = []
    
    for ticker in tickers:
```

File: modules/mean_variance_frontier.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  standard_deviation_returns_list = []
+  weights = [1, 0.9, 0.8, 0.7, 0.6, 0.5, 0.4, 0.3, 0.2, 0.1, 0]
+  number_trading_days = 252
+  daily_returns_df = pd.DataFrame()
+
+  
+  for random_ticker in random_tickers:
+      random_daily_returns_df = pd.DataFrame(portfolio.loc[:, (random_ticker, "daily_returns")])
+      average_returns = portfolio.loc[:, (random_ticker, "daily_returns")].mean() * number_trading_days
+      standard_deviation = (portfolio.loc[:, (random_ticker, "daily_returns")].std()) * np.sqrt(number_trading_days)
```
Comment (mmarovich) [inline modules/mean_variance_frontier.py]:
```suggestion
      annualized_standard_deviation = (portfolio.loc[:, (random_ticker, "daily_returns")].std()) * np.sqrt(number_trading_days)
```

File: modules/mean_variance_frontier.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+      average_returns_list.append(average_returns)
+      standard_deviation_returns_list.append(standard_deviation)
```
Comment (mmarovich) [inline modules/mean_variance_frontier.py]:
```suggestion
      standard_deviation_returns_list.append(annualized_standard_deviation)
```

File: modules/mean_variance_frontier.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
+
+# Function to calculate and plot the Mean Variance Frontier
+
+import random
+import numpy as np
+import pandas as pd
+import seaborn as sns
```
Comment (mmarovich) [inline modules/mean_variance_frontier.py:7]: Delete seaborn, as it is not used.  And also, add:

`from matplotlib.axes._axes import _log as matplotlib_axes_logger
matplotlib_axes_logger.setLevel('ERROR')`

which will get rid of error in app

**Review by mmarovich — CHANGES_REQUESTED** [review]

**Review by mmarovich — APPROVED** [review]

**Pull request closed without merging**